src/components/switchTheme/switchTheme.test.tsx

```diff
@@ -1,53 +1,50 @@
-import { fireEvent, render, screen } from '@testing-library/react';
-import { describe, expect, it, vi } from 'vitest';
-import { SwitchTheme } from 'src/components/switchTheme/switchTheme';
-import { useDispatch, useSelector } from 'react-redux';
-import { selectMenu } from 'src/store/selectors/menuButton';
-import { toggleTheme } from 'src/store/slices/switchTheme';
+import { render } from '@testing-library/react';
+import { useSelector } from 'react-redux';
+import { SwitchTheme } from './switchTheme';
+import { useTheme } from './hook';
+import { ThemeContextProvider } from './useThemeContext';
+import { vi } from 'vitest';
 
 vi.mock('react-redux', () => ({
   useSelector: vi.fn(),
-  useDispatch: vi.fn(),
 }));
 
-vi.mock('src/components/_common/iconicButton/iconicButton', () => ({
-  IconicButton: (props: any) => <button onClick={props.onClickHandler}>{props.children}</button>,
+vi.mock('./hook', () => ({
+  useTheme: vi.fn(),
 }));
 
-vi.mock('src/components/_common/icons/switchThemeIcon', () => ({
-  SwitchThemeIcon: () => <span>Switch Theme Icon</span>,
-}));
-
-const mockDispatch = vi.fn();
-(vi.mocked(useDispatch) as unknown as jest.MockedFunction<typeof useDispatch>).mockReturnValue(mockDispatch);
-
 describe('SwitchTheme Component', () => {
   it('should render SwitchThemeIcon inside IconicButton when menu is active', () => {
-    (vi.mocked(useSelector) as unknown as jest.MockedFunction<typeof useSelector>).mockImplementation(
-      (selector: any) => {
-        if (selector === selectMenu) return true;
-        return null;
-      },
+    (vi.mocked(useTheme) as unknown as jest.MockedFunction<typeof useTheme>).mockReturnValue({
+      theme: 'light',
+      toggleTheme: vi.fn(),
+    });
+    (vi.mocked(useSelector) as unknown as jest.MockedFunction<typeof useSelector>).mockReturnValue(true);
+
+    const { getByRole } = render(
+      <ThemeContextProvider>
+        <SwitchTheme />
+      </ThemeContextProvider>,
     );
 
-    render(<SwitchTheme />);
-
-    expect(screen.getByText('Switch Theme Icon')).toBeInTheDocument();
-    const button = screen.getByRole('button');
-    fireEvent.click(button);
-    expect(mockDispatch).toHaveBeenCalledWith(toggleTheme(expect.anything()));
+    const button = getByRole('button');
+    expect(button).toBeInTheDocument();
   });
 
   it('should not render anything when menu is not active', () => {
-    (vi.mocked(useSelector) as unknown as jest.MockedFunction<typeof useSelector>).mockImplementation(
-      (selector: any) => {
-        if (selector === selectMenu) return false;
-        return null;
-      },
+    (vi.mocked(useTheme) as unknown as jest.MockedFunction<typeof useTheme>).mockReturnValue({
+      theme: 'light',
+      toggleTheme: vi.fn(),
+    });
+    (vi.mocked(useSelector) as unknown as jest.MockedFunction<typeof useSelector>).mockReturnValue(false);
```

Comment:
Hard-coded types are weaknesses. Yes, we can have exclusives. Yes, this one is not a big deal. But I want you to think about such places as non-hard-codable places first. "what would I do if impossible"

Reply:
have you missed this one?